Content.Client/Instruments/InstrumentSystem.cs

```diff
@@ -266,12 +319,24 @@ private void OnMidiEventRx(InstrumentMidiEventEvent midiEv)
 
         instrument.SequenceDelay = Math.Max(instrument.SequenceDelay, delta);
 
-        var currentTick = renderer.SequencerTick;
+        SendMidiEvents(midiEv.MidiEvent, instrument);
+    }
+
+    private void SendMidiEvents(IReadOnlyList<RobustMidiEvent> midiEvents, InstrumentComponent instrument)
+    {
+        if (instrument.Renderer == null)
+        {
+            Log.Warning($"Tried to send Midi events to an instrument without a renderer.");
+            return;
+        }
+
+        var currentTick = instrument.Renderer.SequencerTick;
 
         // ReSharper disable once ForCanBeConvertedToForeach
-        for (uint i = 0; i < midiEv.MidiEvent.Length; i++)
+        for (uint i = 0; i < midiEvents.Count; i++)
```

Comment:
why not just `var i = 0`? you need to cast it to an int the only time you're using it

Reply:
You missed the usage below where I actually use it to add it to an `uint` variable